src/trans/gpu/algor/hicblas_mod.F90

```diff
@@ -63,10 +63,17 @@ SUBROUTINE HIP_SGEMM_BATCHED( &
     TYPE(C_PTR), INTENT(IN), VALUE :: ALLOC
   END SUBROUTINE HIP_SGEMM_BATCHED
 END INTERFACE
+INTERFACE
+  PUBLIC CLEAN_GEMM
```

Comment:
I don't think this is a valid location to specify `PUBLIC`. In fact I think you can just delete this line.